File: src/main/java/lab6/string_list/Node.java
package lab6.string_list;

public class Node {
    private Node prev = null;
    private Node next = null;
    private String value;

    Node(String value) {
        this.value = value;
    }

    public String getValue() {
        return this.value;
    }

    public Node getNext() {
        return next;
    }

    public void setNext(Node next) {
        this.next = next;
    }

    public Node getPrev() {
        return prev;
    }

    public void setPrev(Node prev) {
        this.prev = prev;
    }
}

File: src/main/java/lab7/my_list/Node.kt
package lab7.my_list

class Node<T>(private var value: T? = null) {
    private var prev: Node<T>? = null
    private var next: Node<T>? = null

    fun getValue(): T? {
        return value
    }

    fun getNext(): Node<T>? {
        return next
    }

    fun setNext(next: Node<T>?) {
        this.next = next
    }

    fun getPrev(): Node<T>? {
        return prev
    }

    fun setPrev(prev: Node<T>) {
        this.prev = prev
    }
}
